semigroups/semifp.py

```diff
@@ -64,7 +199,18 @@ def __init__(self, alphabet, rels):
             raise TypeError('the second argument (relations) must be a ' +
                             'list of pairs of strings')
 
-        # TODO parse relations using ^ etc here, as per in ParseRelators in GAP
+        pure_letter_alphabet = True
+        for i in alphabet:
```

Comment:
If the alphabet is not pure then it is assumed that the relations given are already parsed as otherwise it would be ambiguous if the characters "(", ")" and "^" were being used as letters or to denote brackets or powering